Automattic/wp-calypso
# FSE: Fix JS unit tests

_Note: this PR was split from https://github.com/Automattic/wp-calypso/pull/39000_

This PR makes JS unit tests more consistent (in preparation for also having e2e and phpunit tests.) It also fixes several issues with the test suite.

### Changes proposed in this Pull Request
- Fixes an SPT test which broke after #39120. For some reason, it is not breaking on prod, but it is in this branch. That leads me to believe that the config in prod may not be testing everything correctly.
- Renames unit tests to `.test.js` for easier detection (and to be consestent with .spec.js e2e tests)
- Moves config files to `bin/`
- Fixes running js unit tests locally.

### JS unit and UI tests
The unit test command automatically runs test files that:
1. Are in `apps/full-site-editing`
2. Have the extension `.test.js` (or .ts, .tsx, jsx)

To run JS unit/component/ui tests:
```sh
npx lerna run test:js --scope='@automattic/full-site-editing' --stream
```

### Testing instructions
1. Test FSE build should pass
2. From the wp-calypso root, run `npx lerna run test:js --scope='@automattic/full-site-editing' --stream --no-prefix` on the CLI and the tests should pass.
